integrated_channels/sap_success_factors/client.py

```diff
@@ -139,8 +141,19 @@ def create_course_completion(self, user_id, payload):
         Raises:
             HTTPError: if we received a failure response code from SAP SuccessFactors
         """
-        url = self.enterprise_configuration.sapsf_base_url + self.global_sap_config.completion_status_api_path
-        return self._call_post_with_user_override(user_id, url, payload)
+        base_url = self.enterprise_configuration.sapsf_base_url
+
+        if self.enterprise_configuration.prevent_self_submit_grades:
+            return self._call_post_with_session(
+                base_url + self.GENERIC_COURSE_COMPLETION_PATH,
+                payload
```

Comment:
Are we certain that the payload schema is the exact same for these two requests?

Reply:
Well the old endpoint's documentation doesn't appear to exist, but our included fields are named the same as the new API's documented naming scheme. There are some additional fields in the API's endpoint documentation such as instructorName that I cannot tell are optional or not (it doesn't say optional or required).

I will not really have 100% confidence in the fields without being able to hit the API unfortunately.

Reply:
so yeah the existing attribute names are the same and I am not sure if instructorName, cpeHours, comments, price, etc. were existing or not. Their documentation does say that userID is required, so maybe if the rest were actually required, they would also mention that in the description of the fields. I hope?

Reply:
Through my own testing with our SAP tenant which is sort of still accessible via Postman I *think* I figured out that we actually have to specify more fields in this new request vs. the old one.  This is what I ended up with as a payload:
`{
"userID": "ChanA",
"courseID":"HarvardX+CS50x1",
"providerID": "EDX",
"courseCompleted":true,
"completedTimestamp":1538766064479,
"instructorName":"Drayer",
"grade":"Pass",
"totalHours":40,
"creditHours":10,
"contactHours":8,
"cpeHours":2,
"comments":"Testing new OCN API completion events endpoint",
"price":0.00,
"currency":"USD"
}
`

But this is tedious to test/confirm because you only get one shot at creating a learning history record and then you have to pick a new course.  I'm going to spend a couple minutes seeing if I can determine which fields are required more concretlely.

Reply:
OKAY, I managed to figure out a reasonably straightforward way to test this using our BestRun tenant, which is STILL available for our use via Postman (crazy).  I've confirmed that the minimum fields we need to send in order to record a completion event are:

```
{
"completedTimestamp":1538766064479,
"courseCompleted":true,
"userID": "ChanA",
"courseID":"MITx+3.086x",
"providerID": "EDX"
}
```

It seems that everything else is nice to have but not necessarily critical, at least for our tenant.

Reply:
The response I receive when I submit this very basic payload looks like this:

```
{
    "@odata.context": "$metadata#OCNLearningEvents/$entity",
    "@odata.metadataEtag": "W/\"e51c4bea-1e33-44f2-97ef-6ca1c39eb9f0\"",
    "userID": "ChanA",
    "courseID": "MITx+3.086x",
    "providerID": "EDX",
    "courseCompleted": true,
    "completedTimestamp": 1538766064479,
    "instructorName": null,
    "grade": null,
    "totalHours": 0.0,
    "creditHours": 0.0,
    "contactHours": 0.0,
    "cpeHours": 0.0,
    "comments": null,
    "price": 0.0,
    "currency": null
}
```

Reply:
Interesting. I was under the impression our instance was completely inaccessible. Had I known otherwise, could have tested instead of making you do so. Thank you for that confirmation though!

So this confirms the data and format we're sending is still fine. I will double-check handling of the response though.

Reply:
yeah we're not really doing anything with the response unless it's a non-200 status code.

Reply:
So, a small discrepancy that we overlooked during this conversation, and that we are now seeing in a new error message being returned from the SAPSF tenants:  We are sending the value for `courseCompleted` as `"true"` with quotes, however it seems for the new endpoint the value must be sent as `true` without quotes.  

This was a subtle breaking change in the endpoint schema from the previous version that I must have quickly revised during my testing and didn't realize was actually a breaking change -- ie, I probably thought that I just had specified it wrong the first time I tried to submit a payload to the new endpoint.

@binodpant @alex-sheehan-edx we can talk about it more at our upcoming integrations meeting.

Reply:
Error message being captured in our audit logs:  `{"error":{"code":null,"message":"Invalid value for property 'courseCompleted'."}}`

Reply:
I can try to get it to format that correctly. Something about the 2 methods we use for post are formatting it differently.

Reply:
nvm. It seems the new endpoint just expects actual booleans whereas the old one didn't.
And we're doing that override in the base integrated channels class.
https://github.com/edx/edx-enterprise/blob/master/integrated_channels/integrated_channel/models.py#L262

This might be a little trickier to override. Will take a look

Reply:
and yet also here:
https://github.com/edx/edx-enterprise/blob/master/integrated_channels/sap_success_factors/models.py#L301

So we're overriding the payload data that - judging by the comment - was written for SAP.
Ugh